subprojects/docs/src/docs/release/notes.md

```diff
@@ -43,6 +43,60 @@ dependencies {
 
 Long story short, this can be used to model [optional dependencies](https://github.com/gradle/gradle/issues/867)!
 
+## Kotlin DSL
+
+### Kotlin 1.3.21
+
+The embedded Kotlin version has been upgraded to Kotlin 1.3.21.
+
+Please see the [Kotlin 1.3.21 announcement](https://github.com/JetBrains/kotlin/releases/tag/v1.3.21) for details.
+
+### Type-safe accessors in precompiled script plugins
+
+Starting with Gradle 5.3, Kotlin precompiled project script plugins now have type-safe accessors, just like regular project build scripts.
+
+For example, here is how an hypothetical plugin that sets up a Java project according to some convention would be written as a Kotlin precompiled project script plugin in `buildSrc`:
+
+```kotlin
+// buildSrc/src/main/kotlin/my-java-convention.gradle.kts
+plugins {
+    `java-library`
+    checkstyle
+}
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_11
+    targetCompatibility = JavaVersion.VERSION_11
+}
+
+tasks {
+    withType<JavaCompile> {
```

Comment:
:x: `javaCompile { ...` should work as well

Reply:
It works but doesn't do the same, I'll leave it as is for the sake of the example.
And it's demonstrated using `checkstyleMain` below.